lattices/src/ght_lattice.rs

```diff
@@ -0,0 +1,448 @@
+            let (selfonly, mut otheronly) =
+                self.children
+                    .iter()
+                    .fold((false, false), |mut accum, (k, v)| {
+                        if !other.children.contains_key(k) {
+                            accum.0 = true; // selfonly is true
+                            accum
+                        } else {
+                            match v.partial_cmp(other.children.get(k).unwrap()) {
+                                Some(Greater) | None => {
+                                    accum.0 = true; // selfonly is true
+                                    accum
+                                }
+                                Some(Less) => {
+                                    accum.1 = true; // otheronly is true
+                                    accum
+                                }
+                                Some(Equal) => accum, // no changes
+                            }
+                        }
+                    });
+            // now check if other has keys that are missing in self
+            otheronly |= !other.children.keys().all(|k| self.children.contains_key(k));
+
+            if selfonly && otheronly {
+                // unique stuff on both sides: order is incomparable
+                None
+            } else if selfonly && !otheronly {
+                // unique stuff only in self
+                Some(Greater)
+            } else if !selfonly && otheronly {
+                // unique stuff only in other
+                Some(Less)
+            } else {
+                // nothing unique on either side
+                Some(Equal)
+            }
+        }
+    }
+}
+
+impl<Schema, SuffixSchema, Storage> PartialOrd<GhtLeaf<Schema, SuffixSchema, Storage>>
+    for GhtLeaf<Schema, SuffixSchema, Storage>
+where
+    Schema: Eq + Hash + PartialEqVariadic,
+    SuffixSchema: Eq + Hash,
+    Storage: VariadicSet<Schema = Schema> + PartialEq,
+{
+    fn partial_cmp(&self, other: &GhtLeaf<Schema, SuffixSchema, Storage>) -> Option<Ordering> {
+        match self.elements.len().cmp(&other.elements.len()) {
+            Greater => {
+                if other.elements.iter().all(|tup| self.elements.contains(tup)) {
+                    Some(Greater)
+                } else {
+                    None
+                }
+            }
+            Equal => {
+                if self
+                    .elements
+                    .iter()
+                    .all(|head| other.elements.contains(head))
+                {
+                    Some(Equal)
+                } else {
+                    None
+                }
+            }
+            Less => {
+                if self
+                    .elements
+                    .iter()
+                    .all(|head| other.elements.contains(head))
+                {
+                    Some(Less)
+                } else {
+                    None
+                }
+            }
```

Comment:
Interestingly you could collapse these into one case, but don't because it is less readable
```rust
equal_or_less => {
    if self
        .elements
        .iter()
        .all(|head| other.elements.contains(head))
    {
        Some(equal_or_less)
    } else {
        None
    }
}
```